### PR TITLE
Return the query with the findcontrolstate

### PR DESCRIPTION
**Pull request by DesWurstes**
of [D85140](https://phabricator.services.mozilla.com/D85140).

@Snuffleupagus any suggestions?

### CODE REVIEW

File: web/pdf_find_controller.js [hunk 1]
```diff
@@ -736,6 +736,7 @@ class PDFFindController {
       state,
       previous,
       matchesCount: this._requestMatchesCount(),
+      query: this._query,
```
Comment (Snuffleupagus) [inline web/pdf_find_controller.js:739]: Note that this won't actually return *exactly* what the findbar input contains, which may be surprising to the user.
What you actually want here is probably `rawQuery: this._rawQuery,` instead.

File: web/app.js [hunk 2]
```diff
@@ -2438,12 +2438,18 @@ function webViewerUpdateFindMatchesCount({ matchesCount }) {
   }
 }
 
-function webViewerUpdateFindControlState({ state, previous, matchesCount }) {
+function webViewerUpdateFindControlState({
+  state,
+  previous,
+  matchesCount,
+  query,
```
Comment (Snuffleupagus) [inline web/app.js:2445]:
```suggestion
  rawQuery,
```

**Review by Snuffleupagus — CHANGES_REQUESTED** [review]
Also, please note that in the upstream patch you'll need to extend the existing validation to ensure that the new parameter is a string, before including it in the "PDFJS:Parent:updateControlState" message, see https://searchfox.org/mozilla-central/rev/6cc48251bb97600fdf11a5b4c5f621bfc8606d55/toolkit/components/pdfjs/content/PdfStreamConverter.jsm#536-565

**Comment by DesWurstes** [comment]
Thanks!

File: web/pdf_find_controller.js [hunk 3]
```diff
@@ -736,6 +736,7 @@ class PDFFindController {
       state,
       previous,
       matchesCount: this._requestMatchesCount(),
+      rawQuery: this._rawQuery,
```
Comment (Snuffleupagus) [inline web/pdf_find_controller.js:739]: Really sorry, I just looked at this more in-depth and realized that this property is not guaranteed to always be set when `_updateUIState` is called.

The following should be much safer in general:
```suggestion
      rawQuery: this._state ? this._state.query : null,
```
assuming that you also add validation similar to https://searchfox.org/mozilla-central/rev/6cc48251bb97600fdf11a5b4c5f621bfc8606d55/toolkit/components/pdfjs/content/PdfStreamConverter.jsm#554-557, i.e. adding the following in `PdfStreamConverter.jsm` (such that the parameter is optional):
```javascript
const rawQuery = typeof data.rawQuery === "string" ?  data.rawQuery : null;
```

**Review by Snuffleupagus — APPROVED** [review]
r=me, with the final comment addressed.

Sorry about the back-and-forth here, it's slightly more difficult to review these types of patches that need to be split between Bugzilla/Github.

File: web/pdf_find_controller.js (hunk 3 above)
Comment (DesWurstes) [reply web/pdf_find_controller.js]: I have added:

```js
    // Same for the `rawQuery` property.
    let rawQuery = null;
    if (typeof data.rawQuery === "string") {
      rawQuery = data.rawQuery;
    }
```

**Review by timvandermeij — APPROVED** [review]

**Comment by timvandermeij** [comment]
Thanks!

**Pull request merged**